subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/MissingFileSnapshot.java

```diff
@@ -49,4 +53,11 @@ public boolean isContentAndMetadataUpToDate(FileSystemLocationSnapshot other) {
     public void accept(FileSystemSnapshotVisitor visitor) {
         visitor.visitFile(this);
     }
+
+    public boolean isSymbolicLink() {
```

Comment:
We don't need to check for symbolic links except for better error messages. And if it is only for error messages, then we don't need to cache the result. I think the `isSymbolicLink()` should not be on `MissingFileSnapshot`, but when we create an error message we either do it in a general way (`cannot read file ...`) or distinguish between the different kinds there.

Reply:
Reverted: As discussed, we won't yet differentiate between unreadable files in error messages during classpath scanning

Reply:
Reverted